docs/java/features/odata/generate-client.md

```diff
@@ -0,0 +1,270 @@
+    <scope>provided</scope>
+</dependency>
+<dependency>
+    <groupId>javax.inject</groupId>
+    <artifactId>javax.inject</artifactId>
+    <scope>provided</scope>
+</dependency>
+```
+
+</TabItem>
+</Tabs>
+
+Lombok and dependency injections are used by the generated VDM classes, that is why they are needed but only with the scope _provided_.
+
+
+### Using the OData Generator Maven Plugin ###
+
+<Tabs defaultValue="v4" values={[
+{ label: 'OData V2', value: 'v2', },
+{ label: 'OData V4', value: 'v4', }]}>
+
+<TabItem value="v4">
+
+1. Update your `application/pom.xml` file by adding the generator plugin under the `<plugin>` section.
+
+    ```XML
+    <plugin>
+        <groupId>com.sap.cloud.sdk.datamodel</groupId>
+        <artifactId>odata-v4-generator-maven-plugin</artifactId>
+        <!-- Please use the latest version here-->
+        <version>3.18.0</version>
+        <executions>
+            <execution>
+                <id>generate-consumption</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>generate</goal>
+                </goals>
+                <configuration>
+                    <inputDirectory>${project.basedir}/edmx</inputDirectory>
+                    <outputDirectory>${project.build.directory}/vdm</outputDirectory>
+                    <deleteOutputDirectory>true</deleteOutputDirectory>
+                    <packageName>com.mycompany.vdm</packageName>
+                    <defaultBasePath>odata/v4/</defaultBasePath>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+</TabItem>
+<TabItem value="v2">
+
+1. Update your `application/pom.xml` file by adding the generator plugin under the `<plugin>` section.
+
+    ```XML
+    <plugin>
+        <groupId>com.sap.cloud.sdk.datamodel</groupId>
+        <artifactId>odata-generator-maven-plugin</artifactId>
+        <!-- Please use the latest version here-->
+        <version>3.18.0</version>
+        <executions>
+            <execution>
+                <id>generate-consumption</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>generate</goal>
+                </goals>
+                <configuration>
+                    <inputDirectory>${project.basedir}/edmx</inputDirectory>
+                    <outputDirectory>${project.build.directory}/vdm</outputDirectory>
+                    <deleteOutputDirectory>true</deleteOutputDirectory>
+                    <packageName>com.mycompany.vdm</packageName>
+                    <defaultBasePath>odata/v4/</defaultBasePath>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+</TabItem>
+</Tabs>
+
+2. Adapt the `<inputDirectory>` to point to the place of your service definition.
+
+1. In case the target folder should be automatically added as a source folder by maven you can leverage the build helper plugin:
```

Comment:
Generally in the context of directories on your OS, I would always speak of directories. I think folder has a German touch 😊